app/routes.py

```diff
@@ -1,2 +1,208 @@
+        if "completed_at" in form_data:
+            task.completed_at = form_data["completed_at"]
+        db.session.commit() # commiting changes to db
+        return task.to_json_response(), 200
+    return make_response(""), 404
+
+# Delete a task
+@task_bp.route("/<task_id>", methods=["DELETE"], strict_slashes = False)
+def delete_task(task_id):  
+    task = Task.query.get(task_id) 
+    if task != None:
+        db.session.delete(task)
+        db.session.commit()
+        details_str = f"Task {task_id} \"{task.title}\" successfully deleted"
+        return jsonify(details = details_str), 200
+    return make_response(""), 404
+
+# Modify part of a task to mark complete
+@task_bp.route("/<task_id>/mark_complete", methods=["PATCH"], strict_slashes = False)  
+def patch_task_mark_complete(task_id):
+    task = Task.query.get(task_id)
+    # PATCHing if: arg mark_complete specified # in URL and task id 
+    # provided exists in database
+    if task:                         
+        # then call function that changes it to complete
+        task.set_completion() # updates it with a date in "completed_at" field
+        call_slack(task) # insert post for slack here
+        db.session.commit()
+        return task.to_json_response(), 200
+    return make_response(""), 404
+
+def call_slack(task):
+    '''helper function that posts message to slack'''
+    API_KEY = os.environ.get("SLACK_BOT_TOKEN")
+    url = "https://slack.com/api/chat.postMessage"
+    slack_str = f"Someone just completed the task {task.title}"
+    # comes from postman body # ID of the channel you want to send the message to
+    channel_id = "C0211KC1QSK"
+    requests.post(url, data={"token": API_KEY, "channel": channel_id, \
+        "text": slack_str})
+
+# Modify part of a task to mark incomplete
+@task_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"], strict_slashes = False)  
+def patch_task_mark_incomplete(task_id):
+    task = Task.query.get(task_id)
+    if task:
+        task.completed_at = None
+        db.session.commit()
+        return task.to_json_response(), 200
+    return make_response(""), 404
+
+# GOAL ROUTES start here 
+@goal_bp.route("", methods = ["POST"], strict_slashes = False)
+def create_goal():
+    try:
+        request_body = request.get_json()
+        new_goal = Goal(title=request_body["title"])
+        db.session.add(new_goal) # "adds model to the db"
+        db.session.commit() # does the action above
+        return new_goal.goal_to_json_response(), 201
+    except KeyError:
+        return {"details": "Invalid data"}, 400
+
+#Retrieve all /goals 
+@goal_bp.route("", methods = ["GET"], strict_slashes = False)
+def retrieve_goals_data():
+    goals = Goal.query.all() 
+    goals_response = []
+
+    if goals != None:
+        for goal in goals:
+            goals_response.append(goal.simple_response())
+    return jsonify(goals_response), 200 ## OR #return goals_response, 200 
```

Comment:
Consider using `make_response` when you return a 200 level response just as you do for a 400 level response. This consistency adds to readability